### PR TITLE
[Testing library] Update and stabilization of shared testing-libs code.

### PR DESCRIPTION
**Pull request by CesarValiente**
In this branch/PR:
- I've added a `README` file with information about how to use the library and its API.
- Added `SurfaceDuo2` functionality that helps to `span` and `unspan` an app across the whole display area.
- Small refactor on a few comments and headers.

I've also added the version bump to prepare the code for a new release update (alpha2).

**Update: since this testing library will be used for the compose-testing-lib to be built on top, I've made several changes in code and docs in order to provide that functionality with a good API.**

### CODE REVIEW

File: utils/test-utils/README.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+Contains useful specs information and utilities for tests that target **Surface Duo** devices.
+
+`object SurfaceDuo1`
+
+Inside this object you can find useful information about **Surface Duo 1** specs and utilities to span and unspan apps on this device and emulator.
+
+`fun switchFromSingleToDualScreen()`
+
+Span an app across displays. The function will handle correctly the different rotation modes of the device.
+This function works on **Surface Duo 2**.
```
Comment (bimiron) [inline utils/test-utils/README.md:151]: I think here it should also be "Surface Duo **1**"

File: utils/test-utils/README.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            google()
+            mavenCentral()
+         }
+    }
+    ```
+
+2. Add dependencies to the module-level **build.gradle** file (current version may be different from what's shown here):
+
+    ```gradle
+    implementation "com.microsoft.device.dualscreen.testing:testing-kotlin:1.0.0-alpha1"
```
Comment (bimiron) [inline utils/test-utils/README.md:24]: Should we also update this version to 1.0.0-alpha2?

**Review by bimiron — APPROVED** [review]
Nice work!

File: utils/test-utils/README.md (hunk 2 above)
Comment (CesarValiente) [reply utils/test-utils/README.md:24]: totally! good catch!

File: utils/test-utils/README.md (hunk 1 above)
Comment (CesarValiente) [reply utils/test-utils/README.md:151]: wops!! good catch!

**Review by soalb-m — APPROVED** [review]

File: utils/test-utils/README.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+### CurrentActivityDelegate
+
+[Activity](https://developer.android.com/reference/android/app/Activity) delgate used to setup and support a test scenario.
+It provides useful functionality used whenever you want to ensure that the Activity is running before use any view action or assertion on its UI components.
+
+API functions:
+
+`fun <T : Activity> setup(activityScenarioRule: ActivityScenarioRule<T>)`
+
+Setup the delegate using a given ActivityScenarioRule/
```
Comment (soalb-m) [inline utils/test-utils/README.md:40]: There is a "/" at the end. Is it intentional?

File: utils/test-utils/README.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    implementation "com.microsoft.device.dualscreen.testing:testing-kotlin:1.0.0-alpha1"
+    ```
+
+## API reference
+
+The library is divided into a few classes that cover just one specific purpose:
+
+### CurrentActivityDelegate
+
+[Activity](https://developer.android.com/reference/android/app/Activity) delgate used to setup and support a test scenario.
```
Comment (soalb-m) [inline utils/test-utils/README.md:33]: delgate  - typo.

File: utils/test-utils/README.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+API functions:
+
+`fun <T : Activity> setup(activityScenarioRule: ActivityScenarioRule<T>)`
+
+Setup the delegate using a given ActivityScenarioRule/
+
+`fun <T : Activity> clear(activityScenarioRule: ActivityScenarioRule<T>)`
+
+Removes a given ActivityScenarioRule from the delegate
```
Comment (soalb-m) [inline utils/test-utils/README.md:44]: Should end the sentence with a "."

File: utils/test-utils/README.md [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+
+`fun resetActivity()`
+
+Resets the last started Activity to *null*.
+
+`fun resetActivityCounter()`
+
+Resets the activity counter when waiting for the Activity to start before calling it
```
Comment (soalb-m) [inline utils/test-utils/README.md:52]: Should end the sentence with a "."

File: utils/test-utils/README.md [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+Blocks and waits for the next window layout info changes to happen.
+Returns `true` if the window layout info changed before the timeout count reached zero, and `false` if the waiting time elapsed before the changes happened.
+
+`fun reset()`
+
+Resets current used instance to its original state.
+
+## Contributing
+
+Please read the [code of conduct](CODE_OF_CONDUCT.md) and [contribution guidelines](CONTRIBUTING.md).
```
Comment (soalb-m) [inline utils/test-utils/README.md:238]: Both links lead to 404 errors.

**Review by soalb-m — CHANGES_REQUESTED** [review]
There are some syntactic errors and two 404 links.

File: utils/test-utils/README.md (hunk 3 above)
Comment (CesarValiente) [reply utils/test-utils/README.md:40]: 😱 nice catch!

File: utils/test-utils/README.md (hunk 7 above)
Comment (CesarValiente) [reply utils/test-utils/README.md:238]: good catch! Thanks! Will change the `contributing` section entirely.

File: utils/test-utils/README.md (hunk 1 above)
Comment (CesarValiente) [reply utils/test-utils/README.md]: Fixed

File: utils/test-utils/README.md (hunk 2 above)
Comment (CesarValiente) [reply utils/test-utils/README.md]: Fixed

File: utils/test-utils/README.md (hunk 3 above)
Comment (CesarValiente) [reply utils/test-utils/README.md]: Fixed

File: utils/test-utils/README.md (hunk 4 above)
Comment (CesarValiente) [reply utils/test-utils/README.md]: Fixed

File: utils/test-utils/README.md (hunk 6 above)
Comment (CesarValiente) [reply utils/test-utils/README.md]: Fixed

File: utils/test-utils/README.md (hunk 7 above)
Comment (CesarValiente) [reply utils/test-utils/README.md]: Fixed

File: utils/test-utils/README.md (hunk 5 above)
Comment (CesarValiente) [reply utils/test-utils/README.md]: Fixed

File: tabs/tabs/src/androidTest/java/com/microsoft/device/dualscreen/tabs/SurfaceDuoTabLayoutTest.kt [hunk 8]
```diff
@@ -35,66 +36,75 @@ import org.junit.runner.RunWith
...
 
     @Test
     fun testDisplayPositionFromLayout() {
-        SurfaceDuo1.switchFromSingleToDualScreen()
+        uiDevice.spanFromStart()
+        onView(withId(R.id.tabs)).check(matches(areTabsOnScreen(DisplayPosition.DUAL)))
+    }
+
+    @Test
+    fun testCesar() {
```
Comment (bimiron) [inline tabs/tabs/src/androidTest/java/com/microsoft/device/dualscreen/tabs/SurfaceDuoTabLayoutTest.kt:51]: I hope this test never fails 😄 should we rename it?

File: utils/test-utils/README.md [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+
+    ```gradle
+    implementation "com.microsoft.device.dualscreen.testing:testing-kotlin:1.0.0-alpha2"
+    ```
+
+3. Optional. If you need to use other testing dependencies such as [Espresso](https://developer.android.com/training/testing/espresso) or [UiAutomator](https://developer.android.com/training/testing/other-components/ui-automator), you will have to add them to your **build.gradle** file as well. This library uses these dependencies but are not exposed.
+
+# API reference
+
+The library is divided into a few classes that cover just one specific purpose:
```
Comment (bimiron) [inline utils/test-utils/README.md:31]: What is the purpose?

**Review by bimiron — APPROVED** [review]

File: tabs/tabs/src/androidTest/java/com/microsoft/device/dualscreen/tabs/SurfaceDuoTabLayoutTest.kt (hunk 8 above)
Comment (CesarValiente) [reply tabs/tabs/src/androidTest/java/com/microsoft/device/dualscreen/tabs/SurfaceDuoTabLayoutTest.kt:51]: ouchhhhhhhhhh 😅

File: utils/test-utils/README.md (hunk 9 above)
Comment (CesarValiente) [reply utils/test-utils/README.md:31]: wops! no! it shouldn't be there! good catch!

Comment (CesarValiente) [reply utils/test-utils/README.md:31]: Acts as the entry starting point for the following classes API

**Review by soalb-m — APPROVED** [review]

**Pull request merged**